src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -12,14 +12,52 @@ export class WebAssemblyResourceLoader {
       credentials: 'include',
       cache: networkFetchCacheMode
     });
+
+    // While we can expect an ASP.NET Core hosted application to include the environment, other
+    // hosts may not. Assume 'Production' in that case.
+    const environment = bootConfigResponse.headers.get('Blazor_Environment') || 'Production';
```

Comment:
👍 This approach looks great.

Minor nitpick: Isn't it more conventional for HTTP header names to be `Dash-Separated`?